crates/ruff/src/args.rs

```diff
@@ -41,6 +41,32 @@ pub struct Args {
     pub command: Command,
     #[clap(flatten)]
     pub log_level_args: LogLevelArgs,
+    /// Either a path to a TOML configuration file (`pyproject.toml` or `ruff.toml`),
+    /// or a TOML `<KEY> = <VALUE>` pair
+    /// (such as you might find in a `ruff.toml` configuration file)
+    /// overriding a specific configuration option.
+    /// Overrides of individual settings using this option always take precedence
+    /// over all configuration files, including configuration files that were also
+    /// specified using `--config`.
+    #[arg(
+        long,
+        action = clap::ArgAction::Append,
+        value_name = "CONFIG_OPTION",
+        value_parser = ConfigArgumentParser,
+        global = true,
+    )]
+    pub config: Vec<SingleConfigArgument>,
+    /// Ignore all configuration files.
+    //
+    // Note: We can't mark this as conflicting with `--config` here
+    // as `--config` can be used for specifying configuration overrides
+    // as well as configuration files.
+    // Specifying a configuration file conflicts with `--isolated`;
+    // specifying a configuration override does not.
+    // If a user specifies `ruff check --isolated --config=ruff.toml`,
+    // we emit an error later on, after the initial parsing by clap.
+    #[arg(long, help_heading = "Miscellaneous", global = true)]
+    pub isolated: bool,
 }
```

Comment:
I don't have a good suggestion. `ArgGroup` could be what we want but I'm not sure.

What I'm thinking is that it would be nice to group the global settings (potentially even with a different header that these are global options). The benefit of grouping the arguments in a struct is that we could pass them around as a single argument instead of passing each value separately (doesn't scale well)